default/AI/DiplomaticCorp.py

```diff
@@ -98,7 +92,8 @@ def evaluate_diplomatic_attitude(self, other_empire_id):
         num_peace_requests = len(foAI.foAIstate.diplomatic_logs.get('peace_requests', {}).get(log_index, []))
         num_war_declarations = len(foAI.foAIstate.diplomatic_logs.get('war_declarations', {}).get(log_index, []))
         # Too many requests for peace irritate the AI, as do any war declarations
-        irritation = (foAI.foAIstate.aggression * (2.0 + num_peace_requests / 10.0 + 2.0 * num_war_declarations) + 0.5)
+        aggression = foAI.foAIstate.character.get_behavior(Aggression)
```

Comment:
I have 3 instances of `Behavior` here: `character`, `character.get_behavior(Aggression)`, and `character.get_behavior(EmpireIDBehavior)`.  How can I understand which one I should take?  Why we need to have any function composition if we manually choose which behavior to use?